docs/docsite/_ext/build_context.py

```diff
@@ -0,0 +1,110 @@
+"""Sphinx extension for setting up the build settings."""
+
+import subprocess
+from dataclasses import dataclass
+from functools import cache
+from pathlib import Path
+from tomllib import loads as parse_toml_string
+from typing import Literal
+
+from sphinx.application import Sphinx
+from sphinx.util import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+DOCSITE_ROOT_DIR = Path(__file__).parents[1].resolve()
+DOCSITE_EOL_CONFIG_PATH = DOCSITE_ROOT_DIR / 'end_of_life.toml'
+
+
+@dataclass(frozen=True)
```

Comment:
What's the point of a frozen dataclass if it has mutable attributes?

Reply:
It's just a nice default to have.

Reply:
Also, the point of immutable data structures is that it's harder to get into a situation where their contents would be changed suddenly by functions where you pass them. Of course, this doesn't shield us from the mutable nested pieces of data, but it's the first step. It might be good to use some marshalling library in the future.